### PR TITLE
Add get(keyPath:) to JSONObject

### PR DESCRIPTION
**Pull request by hyeongsik-won**
**What this PR does / why we need it**:
[Add get(keyPath:) to JSONObject](https://github.com/yorkie-team/yorkie-ios-sdk/commit/70bc1af1b92bb1eadf2c50a1f76cd0f1e9ddc5da)
- I arbitrarily used `/.^/` for key separator. It should be changed to something suitable.

[Rename YorkieJSONObjectable to JSONObjectable](https://github.com/yorkie-team/yorkie-ios-sdk/commit/838f6db7a6bd82b72a9bed5c71dd762e93d6f490)

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation**:

```docs

```

**Checklist**:
- [ ] Added relevant tests or not required
- [ ] Didn't break anything

### CODE REVIEW

**Review by humdrum — APPROVED** [review]
LGTM!

File: Sources/Document/Json/JSONObject.swift [hunk 1]
```diff
@@ -22,6 +22,12 @@ import Foundation
  */
 @dynamicMemberLookup
 public class JSONObject {
+    private static let keySeparator = "/.^/"
+    private let reservedCharacterForKey = JSONObject.keySeparator
+    private func isValidKey(_ key: String) -> Bool {
+        return key.contains(self.reservedCharacterForKey) == false
```
[the remark this answers is not in the document]
Comment (hyeongsik-won) [reply Sources/Document/Json/JSONObject.swift:28]: An error occurs with the following code.
```swift
private let reservedCharacterForKey = Self.keySeparator 
// Error: Covariant 'Self' type cannot be referenced from a stored property initializer
```

We have another choice to use Self. :)
```swift
private var reservedCharacterForKey: String {
     Self.keySeparator
}
```

[the remark this answers is not in the document]
Comment (humdrum) [reply Sources/Document/Json/JSONObject.swift:28]: Actually, I meant delete reservedCharacterForKey and use Self.keySeparator. If there is no compilation error.

**Review by hackerwins — APPROVED** [review]
Thanks for your contribution.

**Pull request merged**